src/commonMain/kotlin/fr/acinq/lightning/blockchain/electrum/ElectrumClientExtensions.kt

```diff
@@ -0,0 +1,49 @@
+package fr.acinq.lightning.blockchain.electrum
+
+import fr.acinq.bitcoin.ByteVector32
+import fr.acinq.bitcoin.Satoshi
+import fr.acinq.bitcoin.Transaction
+import fr.acinq.lightning.blockchain.fee.FeeratePerKw
+import fr.acinq.lightning.channel.Commitments
+import fr.acinq.lightning.channel.LocalFundingStatus
+import fr.acinq.lightning.transactions.Transactions
+import fr.acinq.lightning.utils.MDCLogger
+import fr.acinq.lightning.utils.sat
+
+
+suspend fun IElectrumClient.getConfirmations(txId: ByteVector32): Int? {
+    val tx = kotlin.runCatching { getTx(txId) }.getOrNull()
+    return tx?.let { getConfirmations(tx) }
+}
+
+suspend fun IElectrumClient.getConfirmations(tx: Transaction): Int? {
+    val scriptHash = ElectrumClient.computeScriptHash(tx.txOut.first().publicKeyScript)
+    val scriptHashHistory = getScriptHashHistory(scriptHash)
+    val item = scriptHashHistory.find { it.txid == tx.txid }
+    val blockHeight = startHeaderSubscription().blockHeight
+    return item?.let { if (item.blockHeight > 0) blockHeight - item.blockHeight + 1 else 0 }
+}
```

Comment:
We could probably return a non-null value here: if anything fails, that means the transaction is not in the blockchain and thus has 0 confirmations?

Reply:
True, but we should document the fact that `null` should be interpreted as `unconfirmed` as well.

Reply:
Improved the comment here: d206a452.